pkg/dmesg/commands.go

```diff
@@ -0,0 +1,146 @@
+package dmesg
+
+import (
+	"context"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+type Commands struct {
```

Comment:
make this private.

dmesg pkg should only expose this (or a very similar) interface and its implementation.
```
type Watcher interface {
    Watch() -> LogLine
    Close() error
}

type LogLine struct {
    Timestamp time.Time
    Level string
    Facility string
    Content string
}
```